### PR TITLE
Issues/285 fix code list interceptor pre handle

### PR DESCRIPTION
**Pull request by btkodakak**
Please review #285.

### CODE REVIEW

File: terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/codelist/CodeListInterceptor.java [hunk 1]
```diff
@@ -85,18 +85,17 @@
...
-    public void postHandle(HttpServletRequest request,
-            HttpServletResponse response, Object handler,
-            ModelAndView modelAndView) {
-
+    public boolean preHandle(HttpServletRequest request, 
+            HttpServletResponse response, Object handler) throws Exception {
+ 
         if (codeLists == null) {
-            return;
+            return true;
```
Comment (btshimizukza) [inline terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/codelist/CodeListInterceptor.java]: This line does not executed. Please modify a `CodeListInterceptorTest#testPostHandle_not_call_afterPropertiesSet()`.

File: terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java [hunk 2]
```diff
@@ -117,22 +117,23 @@ public void tearDown() throws Exception {
...
-    public void testPostHandle_zero() throws Exception {
+    public void testPreHandle_zero() throws Exception {
 
         // do setup.
         testTarget.setApplicationContext(new StaticApplicationContext());
         testTarget.afterPropertiesSet();
 
         // do test.
-        testTarget.postHandle(mockRequest, mockResponse, null, modelAndView);
+        boolean expectedReturnValue = testTarget.preHandle(mockRequest, mockResponse, null);
```
Comment (btshimizukza) [inline terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java]: Please change to a `actualReturnValue`.

File: terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/codelist/CodeListInterceptor.java [hunk 3]
```diff
@@ -85,18 +85,17 @@
...
      * Sets codelist to the attribute of {@link HttpServletRequest}
      * <p>
-     * Sets codelist to the attribute of {@link HttpServletRequest} after the execution of Controller is finished.
+     * Sets codelist to the attribute of {@link HttpServletRequest} before the execution of Controller.
      * </p>
-     * @see org.springframework.web.servlet.handler.HandlerInterceptorAdapter#postHandle(javax.servlet.http.HttpServletRequest,
-     *      javax.servlet.http.HttpServletResponse, java.lang.Object, org.springframework.web.servlet.ModelAndView)
+     * @see org.springframework.web.servlet.handler.HandlerInterceptorAdapter#preHandle(javax.servlet.http.HttpServletRequest,
+     *      javax.servlet.http.HttpServletResponse, java.lang.Object)
      */
```
Comment (btshimizukza) [inline terasoluna-gfw-web/src/main/java/org/terasoluna/gfw/web/codelist/CodeListInterceptor.java]: Please add `@since 5.0.1` as JavaDoc method comment.

**Comment by btshimizukza** [comment]
@btkodakak 

Please confirm review comments.

**Comment by btshimizukza** [comment]
@btkodakak 

And, please remove a unnecessary import  (line.36).

```java
import org.springframework.web.servlet.ModelAndView;
```

**Comment by btkodakak** [comment]
@btshimizukza 
Fixed. Please re-review this.

File: terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java [hunk 4]
```diff
@@ -215,13 +217,13 @@ public void testPostHandle_multi() throws Exception {
...
     @Test
-    public void testPostHandle_not_call_afterPropertiesSet() throws Exception {
+    public void testPreHandle_not_call_afterPropertiesSet() throws Exception {
 
         // do setup.
         // do nothing.
 
         // do test.
-        testTarget.postHandle(mockRequest, mockResponse, null, modelAndView);
+        testTarget.preHandle(mockRequest, mockResponse, null);
```
Comment (btshimizukza) [inline terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java]: Please assert a return value.

Comment (btkodakak) [inline terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java]: Fixed.

**Comment by btshimizukza** [comment]
@btkodakak 

And, please perform some refactoring at the test class.

* remove [unnecessary import](https://github.com/terasolunaorg/terasoluna-gfw/blob/9fb7062309e782fbfcddc47d4d7e51f143a3457c/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java#L50). (Note : this issue is from previous version)

 ```java
 import org.terasoluna.gfw.web.codelist.CodeListInterceptor;
 ```

* remove [unused instance variable](https://github.com/terasolunaorg/terasoluna-gfw/blob/9fb7062309e782fbfcddc47d4d7e51f143a3457c/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java#L85) & [setup logic](https://github.com/terasolunaorg/terasoluna-gfw/blob/9fb7062309e782fbfcddc47d4d7e51f143a3457c/terasoluna-gfw-web/src/test/java/org/terasoluna/gfw/web/codelist/CodeListInterceptorTest.java#L95).

 ```java
 private ModelAndView modelAndView;
 ```
 ```java
 this.modelAndView = new ModelAndView();
 ```

**Comment by btkodakak** [comment]
Fixed. And I already checked other code which was edited by commit #285.

**Comment by btshimizukza** [comment]
Thank you!
Please backport to 5.0.x.

**Pull request merged**